airflow/providers/amazon/aws/executors/ecs/ecs_executor_config.py

```diff
@@ -0,0 +1,111 @@
+
+
+def _fetch_templated_kwargs() -> dict[str, str]:
+    run_task_kwargs_value = conf.get(CONFIG_GROUP_NAME, AllEcsConfigKeys.RUN_TASK_KWARGS, fallback=dict())
+    return json.loads(str(run_task_kwargs_value))
+
+
+def _fetch_explicit_kwargs() -> dict[str, str]:
+    return prune_dict(
+        {key: conf.get(CONFIG_GROUP_NAME, key, fallback=None) for key in RunTaskKwargsConfigKeys()}
+    )
+
+
+def build_task_kwargs() -> dict:
+    # This will put some kwargs at the root of the dictionary that do NOT belong there. However,
+    # the code below expects them to be there and will rearrange them as necessary.
+    task_kwargs = deepcopy(RUN_TASK_KWARG_DEFAULTS)
+    task_kwargs.update(_fetch_templated_kwargs())
+    task_kwargs.update(_fetch_explicit_kwargs())
+
+    # There can only be 1 count of these containers
+    task_kwargs["count"] = 1  # type: ignore
+    # There could be a generic approach to the below, but likely more convoluted then just manually ensuring
+    # the one nested config we need to update is present. If we need to override more options in the future we
+    # should revisit this.
+    if "overrides" not in task_kwargs:
+        task_kwargs["overrides"] = {}  # type: ignore
+    if "containerOverrides" not in task_kwargs["overrides"]:
+        task_kwargs["overrides"]["containerOverrides"] = [{}]  # type: ignore
+    task_kwargs["overrides"]["containerOverrides"][0]["name"] = task_kwargs.pop(  # type: ignore
+        AllEcsConfigKeys.CONTAINER_NAME
+    )
+    # The executor will overwrite the 'command' property during execution. Must always be the first container!
+    task_kwargs["overrides"]["containerOverrides"][0]["command"] = []  # type: ignore
+
+    if any(
+        [
+            subnets := task_kwargs.pop(AllEcsConfigKeys.SUBNETS, None),
+            security_groups := task_kwargs.pop(AllEcsConfigKeys.SECURITY_GROUPS, None),
+            # Surrounding parens are for the walrus operator to function correctly along with the None check
+            (assign_public_ip := task_kwargs.pop(AllEcsConfigKeys.ASSIGN_PUBLIC_IP, None)) is not None,
+        ]
+    ):
+        network_config = prune_dict(
+            {
+                "awsvpcConfiguration": {
+                    "subnets": str(subnets).split(",") if subnets else None,
+                    "securityGroups": str(security_groups).split(",") if security_groups else None,
+                    "assignPublicIp": parse_assign_public_ip(assign_public_ip),
+                }
+            }
+        )
+
+        if "subnets" not in network_config["awsvpcConfiguration"]:
+            raise ValueError("At least one subnet is required to run a task.")
+
+        task_kwargs["networkConfiguration"] = network_config
+
+    task_kwargs = convert_dict_keys_camel_case(task_kwargs)
+
+    try:
+        json.loads(json.dumps(task_kwargs))
+    except JSONDecodeError:
+        raise ValueError(f"AWS ECS Executor config values must be JSON serializable. Got {task_kwargs}")
```

Comment:
overrides/containerOverrides might contain some sensitive information, so we should not expose it in logs
```suggestion
        raise ValueError(f"AWS ECS Executor config values must be JSON serializable.")
```